recipe/gen_patch_json.py

```diff
@@ -808,8 +808,25 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
         if i >= 0:
             deps[i] = "cudatoolkit >=11.2,<12.0a0"
 
-        if "libiconv >=1.15,<1.16.0a0" in deps:
-            _pin_looser(fn, record, "libiconv", upper_bound="1.17.0")
+        if record.get('timestamp', 0) < 1662860143000:
+            if any(dep.startswith("arpack >=3.7") for dep in deps):
+                _pin_looser(fn, record, "arpack", max_pin="x.x")
+            if any(dep.startswith("libiconv >=1") for dep in deps):
+                _pin_looser(fn, record, "libiconv", max_pin="x")
+            if any(dep.startswith("cairo >=1") for dep in deps):
+                _pin_looser(fn, record, "cairo", max_pin="x")
+            if any(dep.startswith("glpk >=5") for dep in deps):
+                _pin_looser(fn, record, "glpk", max_pin="x")
+            if any(dep.startswith("nlopt >=2.7") for dep in deps):
+                _pin_looser(fn, record, "nlopt", max_pin="x.x")
+            if any(dep.startswith("openjpeg >=2.4") for dep in deps):
+                _pin_looser(fn, record, "openjpeg", max_pin="x")
+            if any(dep.startswith("pango >=1.48") for dep in deps):
+                _pin_looser(fn, record, "pango", max_pin="x")
+            if any(dep.startswith("pango >=5.2") for dep in deps):
+                _pin_looser(fn, record, "xz", max_pin="x")
+            if any(dep.startswith("libxml2 >=2.9") for dep in deps):
+                _pin_looser(fn, record, "libxml2", upper_bound="2.11.0")
```

Comment:
@hmaarrfk, since we don't know if libxml2 2.11 will be compatible with 2.10, we need to have an upper bound.

Reply:
Hmm, yeah, i guess i prefer just going through a migration than touching this repo.

https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/3368

Reply:
This isn't entirely correct either, some packages may be built today, after this PR goes into play, with libxml2 2.9 due to some solver inconsistencies.

Reply:
I'm not sure of the specifics of the other pinnings you've loosened, but maybe we should remove the libxml2 for now.

Reply:
I would like to revert this pin as I get failing builds since two weeks (and finally found the source) with the following error:

```
libxmlsec1.so.1: undefined symbol: xmlNanoFTPInit, version LIBXML2_2.4.30
```

For me this sounds like they are not compatible or at least `libxmlsec1` should be exempt from the repodata patch.

Reply:
Maybe we should move this into a new issue?

Reply:
ok, i would support something that would revert this. Is this possible to revert repodata patches?

Reply:
Yes, see https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/329